### PR TITLE
[ch364] Add confirmation to sign out button on popup menu

### PR DESCRIPTION
**Pull request by Jarrydcrawford**

### CODE REVIEW

File: src/layouts/HeaderMenu.tsx [hunk 1]
```diff
@@ -64,34 +85,55 @@ interface PopupMenuProps extends AuthProps {
...
+  history,
+  isLoggedIn,
+  logout,
+}) => {
+  const [showLogOutConfirm, setLogOutConfirm] = React.useState(false);
+
+  const handleDashboardClick = () => {
+    history.push('/dashboard');
+    closeMenu();
+  };
```
Comment (Jarrydcrawford) [inline src/layouts/HeaderMenu.tsx]: I'm really not sure about this. Only added as the Dashboard link is now the only one that should force the menu to close (as the others take you to another page or sign you out). Suggestions welcome.

File: src/layouts/HeaderMenu.tsx [hunk 2]
```diff
@@ -64,34 +85,55 @@ interface PopupMenuProps extends AuthProps {
...
+
   return (
-    <MenuContainer onClick={closeMenu}>
-      <Container>
-        <MenuList>
-          {isLoggedIn ? (
+    <MenuContainer>
+      <MenuListContainer verticalCenter={showLogOutConfirm}>
+        {showLogOutConfirm ? (
+          <LogOutConfirm onCancel={toggleLogOutConfirm} onConfirm={logout} />
```
Comment (harrygr) [inline src/layouts/HeaderMenu.tsx]: Not sure about this, it's not really reusable anywhere else. Wouldn't the standard `window.confirm` suffice?

File: src/layouts/HeaderMenu.tsx [hunk 3]
```diff
@@ -114,3 +156,23 @@ export const HeaderMenu = ({ isLoggedIn, logout }: Props) => {
     </div>
   );
 };
+
+interface LogOutConfirmProps {
+  onCancel: React.MouseEventHandler<HTMLButtonElement>;
+  onConfirm: React.MouseEventHandler<HTMLButtonElement>;
```
Comment (harrygr) [inline src/layouts/HeaderMenu.tsx]: These callback types needn't be so specific. Ultimately they just need to be `() => void`

File: src/layouts/HeaderMenu.tsx [hunk 4]
```diff
@@ -24,6 +28,23 @@ const MenuContainer = styled('div')`
...
+      ? `
+    display: flex;
+    align-items: center;
+    height: calc(100vh - 5.2rem);
+
+    > div {
+      flex: 1 0 auto;
+    }
+  `
+      : ''};
```
Comment (harrygr) [inline src/layouts/HeaderMenu.tsx]: We should be careful adding loads of single-use styled components. These can get unwieldy quickly and result in inconsistent styles throughout the app.

**Pull request closed without merging**